### PR TITLE
fix local storage restore on useRole with preserveURL (#2015)

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot /retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

File: src/test-run/commands/actions.js [hunk 1]
```diff
@@ -324,13 +324,15 @@ export class NavigateToCommand extends Assignable {
...
+        this.storages = null;
 
         this._assignFrom(obj, true);
     }
 
     _getAssignableProperties () {
         return [
-            { name: 'url', type: urlArgument, required: true }
+            { name: 'url', type: urlArgument, required: true },
+            { name: 'storages' }
```
Comment (AlexanderMoskovkin) [inline src/test-run/commands/actions.js]: check the type here. It should be string. In the same way like in other properties in this and other commands

Comment (AlexanderMoskovkin) [reply src/test-run/commands/actions.js]: it should be 'type: nonEmptyStringArgument' without the `required` flag.

**Comment by AlexKamaev** [comment]
FPR

File: src/test-run/commands/validations/argument.js [hunk 2]
```diff
@@ -57,6 +58,15 @@ export function nonEmptyStringArgument (argument, val, createError) {
         throw createError('""');
 }
 
+export function nullableStringArgument (argument, val) {
+    if (val === null) return;
+
+    var type = typeof val;
+
+    if (type !== 'string')
```
Comment (churkin) [inline src/test-run/commands/validations/argument.js]: I more like in one line:
```javascript 
if (typeof val !== 'string' && val !== null)
```

Comment (AlexanderMoskovkin) [reply src/test-run/commands/validations/argument.js]: We use `type` as an argument of the `ActionNullableStringArgumentError` error. There is no reason to call `typeof` twice, so I'd leave this as is

Comment (AlexanderMoskovkin) [reply src/test-run/commands/validations/argument.js]: @churkin Are you agree with my point?

Comment (churkin) [reply src/test-run/commands/validations/argument.js]: @AlexanderMoskovkin  ok, but we can merge conditions:
```javascript
var type = typeof val;

if (type !== 'string' && val !== null)
    ...
```

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by AlexanderMoskovkin** [comment]
FPR /cc @churkin

**Pull request merged**